bin/inx-chronicle/src/collector/mod.rs

```diff
@@ -0,0 +1,275 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::{HashMap, VecDeque};
+
+use async_trait::async_trait;
+use chronicle::{
+    db::{bson::DocError, MongoDb},
+    runtime::{
+        actor::{addr::Addr, context::ActorContext, error::ActorError, event::HandleEvent, report::Report, Actor},
+        config::ConfigureActor,
+        error::RuntimeError,
+    },
+};
+use mongodb::bson::document::ValueAccessError;
+use solidifier::Solidifier;
+use thiserror::Error;
+
+pub mod solidifier;
+
+#[derive(Debug, Error)]
+pub enum CollectorError {
+    #[error(transparent)]
+    Doc(#[from] DocError),
+    #[error(transparent)]
+    MongoDb(#[from] mongodb::error::Error),
+    #[error(transparent)]
+    Runtime(#[from] RuntimeError),
+    #[error(transparent)]
+    ValueAccess(#[from] ValueAccessError),
+}
+
+#[derive(Debug)]
+pub struct Collector {
+    db: MongoDb,
+    solidifier_count: usize,
+}
+
+impl Collector {
+    pub fn new(db: MongoDb, solidifier_count: usize) -> Self {
+        Self { db, solidifier_count }
+    }
+}
+
+#[async_trait]
+impl Actor for Collector {
+    type State = HashMap<usize, Addr<Solidifier>>;
+    type Error = CollectorError;
+
+    async fn init(&mut self, cx: &mut ActorContext<Self>) -> Result<Self::State, Self::Error> {
+        let mut solidifiers = HashMap::new();
+        for i in 0..self.solidifier_count.max(1) {
```

Comment:
I would take the `max` in the `new` ctor. Besides the minimum of `1` should we also define a maximum number of solidifiers as a constant?